### ISSUE TITLE
Internal type checking error on transpile

### ISSUE THREAD

**Issue by rodrigo7491** [informalsystems/apalache#2314]
## Description

When trying to transpile the simple spec below I get an internal type checking error.

## Input specification

```
---------------------------- MODULE ex --------------------------
VARIABLE
    \* @type: Bool;
    val

Init == val = FALSE

Next == val' = ~val
=================================================================
```

## The command line parameters used to run the tool

`apalache-mc transpile ex.tla` leads to the error below
`apalache-mc check ex.tla` works fine

## Expected behavior

File with the generated constraints, as per [ADR-016](https://apalache.informal.systems/docs/adr/016adr-retla.html)

## Error trace

```
PASS #7: WatchdogPass                                             I@11:32:16.562
<unknown>: internal error in type checking: Expected Typed(_: TlaType1), found: Untyped E@11:32:16.575
at.forsyte.apalache.infra.AdaptedException: <unknown>: internal error in type checking: Expected Typed(_: TlaType1), found: Untyped
	at at.forsyte.apalache.infra.passes.PassChainExecutor$.run(PassChainExecutor.scala:39)
	at at.forsyte.apalache.tla.tooling.opt.TranspileCmd.run(TranspileCmd.scala:25)
	at at.forsyte.apalache.tla.Tool$.runCommand(Tool.scala:138)
	at at.forsyte.apalache.tla.Tool$.run(Tool.scala:118)
	at at.forsyte.apalache.tla.Tool$.main(Tool.scala:40)
	at at.forsyte.apalache.tla.Tool.main(Tool.scala)
```

## System information

- Apalache version [`apalache-mc version`]: v0.30.1-44-g0b3aa4e7e
- OS [e.g. Ubuntu Linux or Mac OS, and the current version]: Linux Mint 20.2
- JDK version [e.g. OpenJDK 0.8.0]: openjdk 11.0.17 2022-10-18

## Triage checklist (for maintainers)

- [ ] Reproduce the bug on the main development branch.
- [ ] Add the issue to the apalache GitHub project.
- [ ] If the bug is high impact, ensure someone available is assigned to fix it.

**Comment by shonfeder** [comment]
I discussed this with @Kukovec briefly, and he thinks it's likely that this was introduced by recent integration of the typed builder.

**Comment by Kukovec** [comment]
Slight correction, I said this was the same sort of issue that I've been encountering with the recent rework, I don't think anzthing recent broke it. Transpilation as such is a rather rushed (academic) project, so the tests for it are... lacking.

**Comment by rodrigo7491** [comment]
Sure, thanks for the comments @shonfeder @Kukovec. This academic project might be interesting for some academic people :wink:.

**Comment by thpani** [comment]
Looks like this has been fixed as of latest `v0.43.0-32-gc87725605`
